Content.Shared/Mobs/Systems/MobThresholdSystem.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Shared.Alert;
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Content.Shared.Mobs.Components;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Mobs.Systems;
+
+public sealed class MobThresholdSystem : EntitySystem
+{
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MobThresholdsComponent, MapInitEvent>(MobThresholdMapInit);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentShutdown>(MobThresholdShutdown);
+        SubscribeLocalEvent<MobThresholdsComponent, DamageChangedEvent>(OnDamaged);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentGetState>(OnGetComponentState);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentHandleState>(OnHandleComponentState);
+        SubscribeLocalEvent<MobThresholdsComponent, UpdateMobStateEvent>(OnUpdateMobState);
+    }
+
+    private void OnHandleComponentState(EntityUid uid, MobThresholdsComponent component, ref ComponentHandleState args)
+    {
+        if (args.Current is not MobThresholdComponentState state)
+            return;
+
+        component.Thresholds = new SortedDictionary<FixedPoint2, MobState>(state.Thresholds);
+        component.CurrentThresholdState = state.CurrentThresholdState;
+    }
+
+    private void OnGetComponentState(EntityUid uid, MobThresholdsComponent component, ref ComponentGetState args)
+    {
+        args.State = new MobThresholdComponentState(component.CurrentThresholdState,
+            new Dictionary<FixedPoint2, MobState>(component.Thresholds));
+    }
+
+    private void MobThresholdMapInit(EntityUid uid, MobThresholdsComponent component, MapInitEvent args)
+    {
+        // TODO remove when body sim is implemented
+        EnsureComp<MobStateComponent>(uid);
+        EnsureComp<DamageableComponent>(uid);
```

Comment:
these should just be on the protos instead of ensured in map init

Reply:
Yeah removing this. Originally I did this to minimize downstream YAML changes but @DrSmugleaf convinced me that this is an anti-pattern and it's better to just update the yamls instead.